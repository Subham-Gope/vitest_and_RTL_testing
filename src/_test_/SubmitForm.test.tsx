// import { describe, test, expect, vi } from "vitest";
// import { screen, render, waitFor } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import SubmitForm from "../components/SubmitForm";

// describe("Testing SubmitForm.tsx file", () => {
//   test("inputs data and submits the form", async () => {
//     // Setting up the userEvent
//     const user = userEvent.setup({ delay: 100 });

//     const { container } = render(<SubmitForm />);
//     screen.debug();
//     expect(container).toMatchSnapshot();

//     const name = screen.getByLabelText(/name:/i);
//     const email = screen.getByLabelText(/email:/i);
//     const age = screen.getByLabelText(/^age:$/i);
//     const contactNo = screen.getByLabelText(/contact number/i);
//     const message = screen.getByLabelText(/message/i);
//     const submitBtn = screen.getByRole("button", { name: /send/i });

//     expect(name).toBeInTheDocument();
//     expect(email).toBeInTheDocument();
//     expect(age).toBeInTheDocument();
//     expect(contactNo).toBeInTheDocument();
//     expect(message).toBeInTheDocument();
//     expect(submitBtn).toBeInTheDocument();

//     // Initiating user interaction
//     await user.type(name, "Subham Gope");
//     await user.type(email, "subhamgope@gmail.com");
//     await user.type(age, "24");
//     await user.type(contactNo, "908978787873");
//     await user.type(
//       message,
//       "Hi there, this is a test message, I hope this reaches the test result successfully"
//     );
//     await user.click(submitBtn);

//     expect(name).toHaveValue("Subham Gope");
//     expect(email).toHaveValue("subhamgope@gmail.com");
//     expect(age).toHaveValue(24);
//     expect(contactNo).toHaveValue(908978787873);
//     expect(message).toHaveValue(
//       "Hi there, this is a test message, I hope this reaches the test result successfully"
//     );

//     // Testing the submitted form values

//     // Mocking the global function before rendering
//     const mockFetch = vi.fn().mockResolvedValue({
//       ok: true,
//       json: async () => ({ id: 101 }),
//     });

//     global.fetch = mockFetch;

//     // Wait for the fetch call to happen
//     await waitFor(
//       () => {
//         expect(mockFetch).toHaveBeenCalled();
//       },
//       { timeout: 3000 }
//     );

//     // Get arguments from first fetch call
//     const [url, options] = mockFetch.mock.calls[0];

//     // Assert correct url
//     expect(url).toBe("https://jsonplaceholder.typicode.com/posts");

//     // Assert correct HTTP method and headers
//     expect(options.method).toBe("POST");
//     expect(options.headers["Content-Type"]).toBe("application/json");

//     // Parse and inspect the body sent in fetch
//     const sentBody = JSON.parse(options.body);
//     expect(sentBody).toEqual({
//       contact: {
//         name: "Subham Gope",
//         email: "subhamgope@gmail.com",
//         age: 24,
//         contactNo: "908978787873",
//         message:
//           "Hi there, this is a test message, I hope this reaches the test result successfully",
//       },
//     });
//   });
// });
