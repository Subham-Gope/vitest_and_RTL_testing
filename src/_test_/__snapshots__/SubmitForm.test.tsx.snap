// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing SubmitForm.tsx file > inputs data and submits the form 1`] = `
<div>
  <div
    class="min-h-screen flex items-center justify-center bg-gray-100 p-4"
  >
    <form
      class="w-full max-w-md bg-white shadow-md rounded-2xl p-8 space-y-6"
    >
      <h2
        class="text-2xl font-bold text-center text-gray-800"
      >
        Contact Us
      </h2>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="name"
        >
          Name:
        </label>
        <input
          class="mt-1 w-full px-4 py-2 border border-gray-300 rounded-lg shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500"
          id="name"
          name="name"
          placeholder="John Doe"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="email"
        >
          Email:
        </label>
        <input
          class="mt-1 w-full px-4 py-2 border border-gray-300 rounded-lg shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500"
          id="email"
          name="email"
          placeholder="johndoe@email.com"
          required=""
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="age"
        >
          Age:
        </label>
        <input
          class="mt-1 w-full px-4 py-2 border border-gray-300 rounded-lg shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500"
          id="age"
          name="age"
          placeholder="25"
          required=""
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="contactNo"
        >
          Contact Number:
        </label>
        <input
          class="mt-1 w-full px-4 py-2 border border-gray-300 rounded-lg shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500"
          id="contactNo"
          name="contactNo"
          placeholder="6007844497"
          required=""
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="message"
        >
          Message:
        </label>
        <textarea
          class="mt-1 w-full px-4 py-2 border border-gray-300 rounded-lg shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500"
          id="message"
          name="message"
          placeholder="Type your message..."
          required=""
          rows="4"
        />
      </div>
      <button
        class="w-full py-2 px-4 text-white font-semibold rounded-lg transition bg-blue-600 hover:bg-blue-700"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
</div>
`;
